src/libxrpl/protocol/LedgerFormats.cpp

```diff
@@ -25,347 +25,25 @@ namespace ripple {
-            {sfXChainBridge,             soeREQUIRED},
-            {sfXChainClaimID,            soeREQUIRED},
-            {sfXChainAccountCreateCount, soeREQUIRED},
-            {sfXChainAccountClaimCount,  soeREQUIRED},
-            {sfOwnerNode,                soeREQUIRED},
-            {sfPreviousTxnID,            soeREQUIRED},
-            {sfPreviousTxnLgrSeq,        soeREQUIRED}
-        },
-        commonFields);
-
-    add(jss::XChainOwnedClaimID,
-        ltXCHAIN_OWNED_CLAIM_ID,
-        {
-            {sfAccount,                 soeREQUIRED},
-            {sfXChainBridge,            soeREQUIRED},
-            {sfXChainClaimID,           soeREQUIRED},
-            {sfOtherChainSource,        soeREQUIRED},
-            {sfXChainClaimAttestations, soeREQUIRED},
-            {sfSignatureReward,         soeREQUIRED},
-            {sfOwnerNode,               soeREQUIRED},
-            {sfPreviousTxnID,           soeREQUIRED},
-            {sfPreviousTxnLgrSeq,       soeREQUIRED}
-        },
-        commonFields);
-
-    add(jss::XChainOwnedCreateAccountClaimID,
-        ltXCHAIN_OWNED_CREATE_ACCOUNT_CLAIM_ID,
-        {
-            {sfAccount,                         soeREQUIRED},
-            {sfXChainBridge,                    soeREQUIRED},
-            {sfXChainAccountCreateCount,        soeREQUIRED},
-            {sfXChainCreateAccountAttestations, soeREQUIRED},
-            {sfOwnerNode,                       soeREQUIRED},
-            {sfPreviousTxnID,                   soeREQUIRED},
-            {sfPreviousTxnLgrSeq,               soeREQUIRED}
-        },
-        commonFields);
-
-    add(jss::DID,
-        ltDID,
-        {
-            {sfAccount,              soeREQUIRED},
-            {sfDIDDocument,          soeOPTIONAL},
-            {sfURI,                  soeOPTIONAL},
-            {sfData,          soeOPTIONAL},
-            {sfOwnerNode,            soeREQUIRED},
-            {sfPreviousTxnID,        soeREQUIRED},
-            {sfPreviousTxnLgrSeq,    soeREQUIRED}
-        },
-        commonFields);
-
-    add(jss::Oracle,
-        ltORACLE,
-        {
-            {sfOwner,               soeREQUIRED},
-            {sfProvider,            soeREQUIRED},
-            {sfPriceDataSeries,     soeREQUIRED},
-            {sfAssetClass,          soeREQUIRED},
-            {sfLastUpdateTime,      soeREQUIRED},
-            {sfURI,                 soeOPTIONAL},
-            {sfOwnerNode,           soeREQUIRED},
-            {sfPreviousTxnID,       soeREQUIRED},
-            {sfPreviousTxnLgrSeq,   soeREQUIRED}
-        },
-        commonFields);
-
-    // clang-format on
+#pragma push_macro("UNWRAP")
+#undef UNWRAP
+#define UNWRAP(...) __VA_ARGS__
+#pragma push_macro("OBJECT")
+#undef OBJECT
+#define OBJECT(tag, value, name, fields) \
+    add(jss::name, tag, UNWRAP fields, commonFields);
+#include <xrpl/protocol/objects.h>
+#undef OBJECT
+#pragma pop_macro("OBJECT")
+#undef UNWRAP
+#pragma pop_macro("UNWRAP")
```

Comment:
For consistency, please follow the same style as in Feature.cpp (vertical spacing)